xarray/backends/h5netcdf_.py

```diff
@@ -88,6 +88,20 @@ def _h5netcdf_create_group(dataset, name):
     return dataset.create_group(name)
 
 
+def _h5netcdf_opener(filename, mode, storage_options=None, **kwargs):
+    import h5netcdf
+
+    if isinstance(filename, str) and is_remote_uri(filename):
+        import fsspec
+
+        mode_ = "rb" if mode == "r" else mode
+        fs, _, _ = fsspec.get_fs_token_paths(
+            filename, mode=mode_, storage_options=storage_options
+        )
+        filename = fs.open(filename, mode=mode_)
```

Comment:
Not an expert in this bit but there's a `_find_absolute_path` in `backends/common.py` that shares a lot of code with the first three lines here.

Reply:
Do we really want to unconditionally open remote urls with `fsspec`? This contradicts the usage of native implementations (via "driver"-kwarg, see #8360) in h5py/hdf5.

Reply:
> Do we really want to unconditionally open remote urls with fsspec?

My guess is yes by default. That's what other pydata libraries like pandas, Dask, Zarr, etc. have converged on for file handling, so it would be familiar to many users. That said, if `driver=` offers some benefits over `fsspec` (again, I'm not familiar with the new `driver` functionality) it'd be easy for these two approaches to live alongside each other:

- Use `fsspec` by default is no `driver` is specified
- If `driver` is specified use that instead
- If there are conflicting options provided for some reason (e.g. `driver=` and `storage_options=`) then raise an informative error.

Reply:
The algorithm shown here looks good to me. I also think fsspec is more used by users although keeping the ros3 alternative is desirable too (see [this](https://pynwb.readthedocs.io/en/stable/tutorials/advanced_io/streaming.html#which-streaming-method-to-choose)).

Reply:
These changes looks indeed very similar to `_find_absolute_paths`, can we instead get that one working for this case as well? 
https://github.com/pydata/xarray/blob/49bd63a8332c1930a866724a2968b2d880dae25e/xarray/backends/common.py#L60-L115